File: 2. Structural Design Patterns/Adapter/Kotlin/AdapterTemplate.kt
// Adapter Pattern – Kotlin Template
// Target expects request(); Adaptee has specificRequest(); Adapter bridges them.

interface Target { fun request() }
class Adaptee { fun specificRequest() { /* ... */ } }
class Adapter(private val adaptee: Adaptee = Adaptee()) : Target { override fun request() = adaptee.specificRequest() }

fun main(){ val t: Target = Adapter(); t.request() }

File: 2. Structural Design Patterns/Adapter/Javascript/AdapterTemplate.js
// Adapter Pattern – JavaScript Template
// Target expects request(); Adaptee has specificRequest(); Adapter bridges them.

class Target {
  request() {}
}

class Adaptee {
  specificRequest() { /* ... */ }
}

class Adapter extends Target {
  constructor() {
    super();
    this.adaptee = new Adaptee();
  }
  request() {
    this.adaptee.specificRequest();
  }
}

// Usage: const t = new Adapter(); t.request();
